### PR TITLE
Make the entirely selected merged cells (when selecting entire rows/columns) have the same background as a selected area

### PR DESCRIPTION
**Pull request by jansiegel**
Make the entirely selected merged cells (when selecting entire rows/columns) have the same background as a selected area using newly-added hooks: `afterDrawSelection` and `beforeRemoveCellClassNames`.

### Context
#4860

### How has this been tested?
- Added test cases

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature or improvement (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

File: src/pluginHooks.js [hunk 1]
```diff
@@ -219,6 +219,23 @@ const REGISTERED_HOOKS = [
...
+   * @returns {String|undefined} Can return a `String`, which will act as an additional `className` to be added to the currently processed cell.
+   */
+  'afterDrawSelection',
+  /**
+   * Fired inside the Walkontable's `refreshSelections` method. Can be used to remove additional class names from all cells in the table.
+   *
+   * @event Hooks#beforeRemoveCellClassNames
+   * @returns {String[]|undefined} Can return an `Array` of `String`s. Each of these strings will act like class names to be removed from all the cells in the table.
+   */
+  'beforeRemoveCellClassNames',
```
Comment (budnix) [inline src/pluginHooks.js]: Can you add `@since` tag to the newly added hooks?

File: src/plugins/mergeCells/mergeCells.js [hunk 2]
```diff
@@ -859,6 +861,27 @@ class MergeCells extends BasePlugin {
...
+  }
+
+  /**
+   * `beforeRemoveCellClassNames` hook callback. Used to remove additional class name from all cells in the table.
+   *
+   * @returns {String[]} An `Array` of `String`s. Each of these strings will act like class names to be removed from all the cells in the table.
+   */
+  onBeforeRemoveCellClassNames() {
+    return this.selectionCalculations.getSelectedMergedCellClassNameToRemove();
+  }
```
Comment (budnix) [inline src/plugins/mergeCells/mergeCells.js]: Can you mark this 2 hook listeners as `@private`?

File: src/plugins/mergeCells/test/selection.e2e.js [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    hot.selectColumns(0, 2);
+
+    let mergedCell = getCell(0, 0);
+    expect(getComputedStyle(mergedCell, ':before').backgroundColor).toEqual(selectedCellBackground);
+
+    hot.selectRows(0, 2);
+
+    mergedCell = getCell(0, 0);
+    expect(getComputedStyle(mergedCell, ':before').backgroundColor).toEqual(selectedCellBackground);
+  });
```
Comment (budnix) [inline src/plugins/mergeCells/test/selection.e2e.js]: Can I suggest to reformat the tests to make them more "eye" friendly? I mean to group test logic from test expectations, Such as:
```js
...
let mergedCell = getCell(0, 0);

expect(getComputedStyle(mergedCell, ':before').backgroundColor).toEqual(selectedCellBackground);

hot.selectRows(0, 2);
mergedCell = getCell(0, 0);

expect(getComputedStyle(mergedCell, ':before').backgroundColor).toEqual(selectedCellBackground);
...
```
Adding new lines before `expect` make the tests more pleasant to analyze :)

**Comment by budnix** [comment]
I've found an issue related to this where the second selection layer behaves as 0-layer before that fixes. This should be fixed in this PR?

kapture 2018-03-15 at 8 54 29

**Comment by jansiegel** [comment]
The problem from the gif should be fixed now. There are some additional problems with the merged cells/non-contiguous selection, but I think they're more connected with #4859.

**Pull request merged**